apps/web/src/components/layout/components/SideNav.tsx

```diff
@@ -17,29 +19,18 @@ export function SideNav({}: Props) {
   const { setEnvironment, isLoading, environment, readonly } = useEnvController();
   const { currentUser } = useContext(AuthContext);
   const location = useLocation();
+  const { pathname } = location;
   const [opened, setOpened] = useState(readonly);
   const { colorScheme } = useMantineColorScheme();
   const dark = colorScheme === 'dark';
   const { addItem } = useContext(SpotlightContext);
 
   useEffect(() => {
     setOpened(readonly);
-    if (readonly && location.pathname === '/changes') {
+    if (readonly && pathname === '/changes') {
       navigate('/');
     }
-  }, [readonly]);
-
-  useEffect(() => {
-    addItem([
-      {
-        id: 'toggle-environment',
-        title: `Toggle to ${environment?.name === 'Production' ? 'Development' : 'Production'} environment`,
-        onTrigger: () => {
-          setEnvironment(environment?.name === 'Production' ? 'Development' : 'Production');
-        },
-      },
-    ]);
-  }, [environment]);
```

Comment:
The use of `addItem` caused infinite loop re-renders. As it is part of Spotlight shortcuts I moved it to the component to group them all there and manage them.